### DISCUSSION TITLE
Trying to install wine64 for box64 to run a Windows app on RPi4, Examples script failing

### DISCUSSION THREAD

**Discussion by Avasam** [ptitSeb/box64#1463] [Q&A]
Hi! I've been trying to follow the install guide at https://github.com/ptitSeb/box64/blob/main/docs/X64WINE.md , there's a lot of steps and I'm not sure what all of them want me to do. But under Examples, I tried running the script (copied into a file, set as executable, then ran in a shell) and I get the following error:

```
./installing_wine.sh
./installing_wine.sh: line 10: local: can only be used in a function
./installing_wine.sh: line 11: local: can only be used in a function
./installing_wine.sh: line 12: local: can only be used in a function
./installing_wine.sh: line 13: local: can only be used in a function
./installing_wine.sh: line 14: local: can only be used in a function
Downloading wine . . .
Extracting wine . . .
dpkg-deb: error: failed to read archive 'wine--amd64_~_amd64.deb': No such file or directory
dpkg-deb: error: failed to read archive 'wine-_~_amd64.deb': No such file or directory
dpkg-deb: error: failed to read archive 'wine--i386_~_i386.deb': No such file or directory
Installing wine . . .
mv: cannot stat 'wine-installer/opt/wine*': No such file or directory
Hit:1 http://deb.debian.org/debian bookworm InRelease
Hit:2 http://deb.debian.org/debian-security bookworm-security InRelease
Hit:3 http://deb.debian.org/debian bookworm-updates InRelease
Hit:4 http://archive.raspberrypi.com/debian bookworm InRelease
Reading package lists... Done                              
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Note, selecting 'libglu1-mesa:armhf' instead of 'libglu1:armhf'
Package libldap-2.4-2:armhf is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source
However the following packages replace it:
  libldap-common

E: Package 'libldap-2.4-2:armhf' has no installation candidate
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Package libldap-2.4-2 is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source
However the following packages replace it:
  libldap-common

E: Package 'libldap-2.4-2' has no installation candidate
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
libstb0 is already the newest version (0.0~git20220908.8b5f1f3+ds-1).
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
--2024-04-24 20:00:47--  http://ftp.us.debian.org/debian/pool/main/f/faudio/
Resolving ftp.us.debian.org (ftp.us.debian.org)... 64.50.233.100, 64.50.236.52, 208.80.154.139, ...
Connecting to ftp.us.debian.org (ftp.us.debian.org)|64.50.233.100|:80... connected.
HTTP request sent, awaiting response... 200 OK
Length: unspecified [text/html]
Saving to: ‘index.html.tmp’

index.html.tmp          [ <=>                ]  28.25K  --.-KB/s    in 0.05s   

2024-04-24 20:00:48 (514 KB/s) - ‘index.html.tmp’ saved [28924]

Loading robots.txt; please ignore errors.
--2024-04-24 20:00:48--  http://ftp.us.debian.org/robots.txt
Reusing existing connection to ftp.us.debian.org:80.
HTTP request sent, awaiting response... 404 Not Found
2024-04-24 20:00:48 ERROR 404: Not Found.

Removing index.html.tmp since it should be rejected.

--2024-04-24 20:00:48--  http://ftp.us.debian.org/debian/pool/main/f/faudio/?C=N;O=D
Reusing existing connection to ftp.us.debian.org:80.
HTTP request sent, awaiting response... 200 OK
Length: unspecified [text/html]
Saving to: ‘index.html?C=N;O=D.tmp’

index.html?C=N;O=D.     [ <=>                ]  28.25K  --.-KB/s    in 0.005s  

2024-04-24 20:00:48 (5.91 MB/s) - ‘index.html?C=N;O=D.tmp’ saved [28924]

Removing index.html?C=N;O=D.tmp since it should be rejected.

--2024-04-24 20:00:48--  http://ftp.us.debian.org/debian/pool/main/f/faudio/?C=M;O=A
Reusing existing connection to ftp.us.debian.org:80.
HTTP request sent, awaiting response... 200 OK
Length: unspecified [text/html]
Saving to: ‘index.html?C=M;O=A.tmp’

index.html?C=M;O=A.     [ <=>                ]  28.25K  --.-KB/s    in 0.002s  

2024-04-24 20:00:48 (15.2 MB/s) - ‘index.html?C=M;O=A.tmp’ saved [28924]

Removing index.html?C=M;O=A.tmp since it should be rejected.

--2024-04-24 20:00:48--  http://ftp.us.debian.org/debian/pool/main/f/faudio/?C=S;O=A
Reusing existing connection to ftp.us.debian.org:80.
HTTP request sent, awaiting response... 200 OK
Length: unspecified [text/html]
Saving to: ‘index.html?C=S;O=A.tmp’

index.html?C=S;O=A.     [ <=>                ]  28.25K  --.-KB/s    in 0.002s  

2024-04-24 20:00:48 (16.2 MB/s) - ‘index.html?C=S;O=A.tmp’ saved [28924]

Removing index.html?C=S;O=A.tmp since it should be rejected.

--2024-04-24 20:00:48--  http://ftp.us.debian.org/debian/pool/main/f/faudio/?C=D;O=A
Reusing existing connection to ftp.us.debian.org:80.
HTTP request sent, awaiting response... 200 OK
Length: unspecified [text/html]
Saving to: ‘index.html?C=D;O=A.tmp’

index.html?C=D;O=A.     [ <=>                ]  28.25K  --.-KB/s    in 0.002s  

2024-04-24 20:00:48 (14.5 MB/s) - ‘index.html?C=D;O=A.tmp’ saved [28924]

Removing index.html?C=D;O=A.tmp since it should be rejected.

--2024-04-24 20:00:48--  http://ftp.us.debian.org/debian/pool/main/f/faudio/libfaudio0_21.02-1~bpo10+1_i386.deb
Reusing existing connection to ftp.us.debian.org:80.
HTTP request sent, awaiting response... 200 OK
Length: 148616 (145K) [application/vnd.debian.binary-package]
Saving to: ‘libfaudio0_21.02-1~bpo10+1_i386.deb’

libfaudio0_21.02-1~ 100%[===================>] 145.13K  --.-KB/s    in 0.1s    

2024-04-24 20:00:48 (1.15 MB/s) - ‘libfaudio0_21.02-1~bpo10+1_i386.deb’ saved [148616/148616]

FINISHED --2024-04-24 20:00:48--
Total wall clock time: 0.7s
Downloaded: 6 files, 286K in 0.2s (1.48 MB/s)
./
./usr/
./usr/lib/
./usr/lib/i386-linux-gnu/
./usr/lib/i386-linux-gnu/libFAudio.so.0.21.02
./usr/share/
./usr/share/doc/
./usr/share/doc/libfaudio0/
./usr/share/doc/libfaudio0/README
./usr/share/doc/libfaudio0/changelog.Debian.gz
./usr/share/doc/libfaudio0/copyright
./usr/lib/i386-linux-gnu/libFAudio.so.0
'libfaudio/usr/lib/i386-linux-gnu/libFAudio.so.0.21.02' -> '/usr/lib/i386-linux-gnu/libFAudio.so.0.21.02'
removed '/usr/lib/i386-linux-gnu/libFAudio.so.0'
'libfaudio/usr/lib/i386-linux-gnu/libFAudio.so.0' -> '/usr/lib/i386-linux-gnu/libFAudio.so.0'
'libfaudio/usr/share/doc/libfaudio0/README' -> '/usr/share/doc/libfaudio0/README'
'libfaudio/usr/share/doc/libfaudio0/changelog.Debian.gz' -> '/usr/share/doc/libfaudio0/changelog.Debian.gz'
'libfaudio/usr/share/doc/libfaudio0/copyright' -> '/usr/share/doc/libfaudio0/copyright'
chmod: cannot operate on dangling symlink '/usr/local/bin/wine'
chmod: cannot operate on dangling symlink '/usr/local/bin/wine64'
chmod: cannot operate on dangling symlink '/usr/local/bin/wineboot'
chmod: cannot operate on dangling symlink '/usr/local/bin/winecfg'
chmod: cannot operate on dangling symlink '/usr/local/bin/wineserver'

```

**Comment by gitmanbloggus** [comment]
> Hi! I've been trying to follow the install guide at https://github.com/ptitSeb/box64/blob/main/docs/X64WINE.md , there's a lot of steps and I'm not sure what all of them want me to do. But under Examples, I tried running the script (copied into a file, set as executable, then ran in a shell) and I get the following error:
> 
> ```
> ./installing_wine.sh
> ./installing_wine.sh: line 10: local: can only be used in a function
> ./installing_wine.sh: line 11: local: can only be used in a function
> ./installing_wine.sh: line 12: local: can only be used in a function
> ./installing_wine.sh: line 13: local: can only be used in a function
> ./installing_wine.sh: line 14: local: can only be used in a function
> Downloading wine . . .
> Extracting wine . . .
> dpkg-deb: error: failed to read archive 'wine--amd64_~_amd64.deb': No such file or directory
> dpkg-deb: error: failed to read archive 'wine-_~_amd64.deb': No such file or directory
> dpkg-deb: error: failed to read archive 'wine--i386_~_i386.deb': No such file or directory
> Installing wine . . .
> mv: cannot stat 'wine-installer/opt/wine*': No such file or directory
> Hit:1 http://deb.debian.org/debian bookworm InRelease
> Hit:2 http://deb.debian.org/debian-security bookworm-security InRelease
> Hit:3 http://deb.debian.org/debian bookworm-updates InRelease
> Hit:4 http://archive.raspberrypi.com/debian bookworm InRelease
> Reading package lists... Done                              
> Reading package lists... Done
> Building dependency tree... Done
> Reading state information... Done
> Note, selecting 'libglu1-mesa:armhf' instead of 'libglu1:armhf'
> Package libldap-2.4-2:armhf is not available, but is referred to by another package.
> This may mean that the package is missing, has been obsoleted, or
> is only available from another source
> However the following packages replace it:
>   libldap-common
> 
> E: Package 'libldap-2.4-2:armhf' has no installation candidate
> Reading package lists... Done
> Building dependency tree... Done
> Reading state information... Done
> Package libldap-2.4-2 is not available, but is referred to by another package.
> This may mean that the package is missing, has been obsoleted, or
> is only available from another source
> However the following packages replace it:
>   libldap-common
> 
> E: Package 'libldap-2.4-2' has no installation candidate
> Reading package lists... Done
> Building dependency tree... Done
> Reading state information... Done
> libstb0 is already the newest version (0.0~git20220908.8b5f1f3+ds-1).
> 0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
> --2024-04-24 20:00:47--  http://ftp.us.debian.org/debian/pool/main/f/faudio/
> Resolving ftp.us.debian.org (ftp.us.debian.org)... 64.50.233.100, 64.50.236.52, 208.80.154.139, ...
> Connecting to ftp.us.debian.org (ftp.us.debian.org)|64.50.233.100|:80... connected.
> HTTP request sent, awaiting response... 200 OK
> Length: unspecified [text/html]
> Saving to: ‘index.html.tmp’
> 
> index.html.tmp          [ <=>                ]  28.25K  --.-KB/s    in 0.05s   
> 
> 2024-04-24 20:00:48 (514 KB/s) - ‘index.html.tmp’ saved [28924]
> 
> Loading robots.txt; please ignore errors.
> --2024-04-24 20:00:48--  http://ftp.us.debian.org/robots.txt
> Reusing existing connection to ftp.us.debian.org:80.
> HTTP request sent, awaiting response... 404 Not Found
> 2024-04-24 20:00:48 ERROR 404: Not Found.
> 
> Removing index.html.tmp since it should be rejected.
> 
> --2024-04-24 20:00:48--  http://ftp.us.debian.org/debian/pool/main/f/faudio/?C=N;O=D
> Reusing existing connection to ftp.us.debian.org:80.
> HTTP request sent, awaiting response... 200 OK
> Length: unspecified [text/html]
> Saving to: ‘index.html?C=N;O=D.tmp’
> 
> index.html?C=N;O=D.     [ <=>                ]  28.25K  --.-KB/s    in 0.005s  
> 
> 2024-04-24 20:00:48 (5.91 MB/s) - ‘index.html?C=N;O=D.tmp’ saved [28924]
> 
> Removing index.html?C=N;O=D.tmp since it should be rejected.
> 
> --2024-04-24 20:00:48--  http://ftp.us.debian.org/debian/pool/main/f/faudio/?C=M;O=A
> Reusing existing connection to ftp.us.debian.org:80.
> HTTP request sent, awaiting response... 200 OK
> Length: unspecified [text/html]
> Saving to: ‘index.html?C=M;O=A.tmp’
> 
> index.html?C=M;O=A.     [ <=>                ]  28.25K  --.-KB/s    in 0.002s  
> 
> 2024-04-24 20:00:48 (15.2 MB/s) - ‘index.html?C=M;O=A.tmp’ saved [28924]
> 
> Removing index.html?C=M;O=A.tmp since it should be rejected.
> 
> --2024-04-24 20:00:48--  http://ftp.us.debian.org/debian/pool/main/f/faudio/?C=S;O=A
> Reusing existing connection to ftp.us.debian.org:80.
> HTTP request sent, awaiting response... 200 OK
> Length: unspecified [text/html]
> Saving to: ‘index.html?C=S;O=A.tmp’
> 
> index.html?C=S;O=A.     [ <=>                ]  28.25K  --.-KB/s    in 0.002s  
> 
> 2024-04-24 20:00:48 (16.2 MB/s) - ‘index.html?C=S;O=A.tmp’ saved [28924]
> 
> Removing index.html?C=S;O=A.tmp since it should be rejected.
> 
> --2024-04-24 20:00:48--  http://ftp.us.debian.org/debian/pool/main/f/faudio/?C=D;O=A
> Reusing existing connection to ftp.us.debian.org:80.
> HTTP request sent, awaiting response... 200 OK
> Length: unspecified [text/html]
> Saving to: ‘index.html?C=D;O=A.tmp’
> 
> index.html?C=D;O=A.     [ <=>                ]  28.25K  --.-KB/s    in 0.002s  
> 
> 2024-04-24 20:00:48 (14.5 MB/s) - ‘index.html?C=D;O=A.tmp’ saved [28924]
> 
> Removing index.html?C=D;O=A.tmp since it should be rejected.
> 
> --2024-04-24 20:00:48--  http://ftp.us.debian.org/debian/pool/main/f/faudio/libfaudio0_21.02-1~bpo10+1_i386.deb
> Reusing existing connection to ftp.us.debian.org:80.
> HTTP request sent, awaiting response... 200 OK
> Length: 148616 (145K) [application/vnd.debian.binary-package]
> Saving to: ‘libfaudio0_21.02-1~bpo10+1_i386.deb’
> 
> libfaudio0_21.02-1~ 100%[===================>] 145.13K  --.-KB/s    in 0.1s    
> 
> 2024-04-24 20:00:48 (1.15 MB/s) - ‘libfaudio0_21.02-1~bpo10+1_i386.deb’ saved [148616/148616]
> 
> FINISHED --2024-04-24 20:00:48--
> Total wall clock time: 0.7s
> Downloaded: 6 files, 286K in 0.2s (1.48 MB/s)
> ./
> ./usr/
> ./usr/lib/
> ./usr/lib/i386-linux-gnu/
> ./usr/lib/i386-linux-gnu/libFAudio.so.0.21.02
> ./usr/share/
> ./usr/share/doc/
> ./usr/share/doc/libfaudio0/
> ./usr/share/doc/libfaudio0/README
> ./usr/share/doc/libfaudio0/changelog.Debian.gz
> ./usr/share/doc/libfaudio0/copyright
> ./usr/lib/i386-linux-gnu/libFAudio.so.0
> 'libfaudio/usr/lib/i386-linux-gnu/libFAudio.so.0.21.02' -> '/usr/lib/i386-linux-gnu/libFAudio.so.0.21.02'
> removed '/usr/lib/i386-linux-gnu/libFAudio.so.0'
> 'libfaudio/usr/lib/i386-linux-gnu/libFAudio.so.0' -> '/usr/lib/i386-linux-gnu/libFAudio.so.0'
> 'libfaudio/usr/share/doc/libfaudio0/README' -> '/usr/share/doc/libfaudio0/README'
> 'libfaudio/usr/share/doc/libfaudio0/changelog.Debian.gz' -> '/usr/share/doc/libfaudio0/changelog.Debian.gz'
> 'libfaudio/usr/share/doc/libfaudio0/copyright' -> '/usr/share/doc/libfaudio0/copyright'
> chmod: cannot operate on dangling symlink '/usr/local/bin/wine'
> chmod: cannot operate on dangling symlink '/usr/local/bin/wine64'
> chmod: cannot operate on dangling symlink '/usr/local/bin/wineboot'
> chmod: cannot operate on dangling symlink '/usr/local/bin/winecfg'
> chmod: cannot operate on dangling symlink '/usr/local/bin/wineserver'
> 
> ```

pi-apps works better
try that
